system/gatekeeper/templates/constrainttemplate-outdated-image-bases.yaml

```diff
@@ -11,24 +11,39 @@ spec:
     spec:
       names:
         kind: GkOutdatedImageBases
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          properties:
+            doopImageCheckerURL:
+              type: string
 
   targets:
     - target: admission.k8s.gatekeeper.sh
       rego: |
         package outdatedimagebases
 
+        # check each pod container
+        obj := input.review.object
+        container := obj.spec.containers[_]
```

Comment:
I see why you're pulling this out of the `violation[]{ ... }` block, but this won't work as far as I can tell. My understanding would be that this will pick one `container` at random and not consider the rest. You will need to construct an object on the top-level that iterates over all containers and then contains all the responses within itself. Something like:

```rego
checks[{ "image": container.image, "response": resp }] {
  container := input.review.object.spec.containers[_]
  resp := ...
}
```

Then you can iterate over that object in the `violation` blocks.